feat: 게시글 CRUD API 추가

cabbage556 opened the pull request:
createBoard
  - POST '/boards' 라우트 핸들러
  - 생성한 게시글 정보 반환

findAll
  - GET '/boards?page=:page' 라우트 핸들러
  - 페이지에 해당하는 조회 게시글 10개 반환

findOne
  - GET '/boards/:id' 라우트 핸들러
  - 조회한 게시글 정보 반환

updateBoard
  - PATCH '/boards/:id' 라우트 핸들러
  - 업데이트한 게시글 정보 반환

deleteBoard
  - DELETE '/boards/:id' 라우트 핸들러

feat-#9

cabbage556 committed:
feat: 게시글 CRUD API 추가

createBoard
  - POST '/boards' 라우트 핸들러
  - 생성한 게시글 정보 반환

findAll
  - GET '/boards?page=:page' 라우트 핸들러
  - 페이지에 해당하는 조회 게시글 10개 반환

findOne
  - GET '/boards/:id' 라우트 핸들러
  - 조회한 게시글 정보 반환

updateBoard
  - PATCH '/boards/:id' 라우트 핸들러
  - 업데이트한 게시글 정보 반환

deleteBoard
  - DELETE '/boards/:id' 라우트 핸들러

feat-#9

cabbage556 commented:
유저 권한 확인 기능은 추가하지 않고 기본 CRUD API 형태만 구현하였음

HUMBLE25 commented on src/apis/boards/boards.service.ts:
```diff
@@ -0,0 +1,91 @@
+			title: createBoardDTO.title,
+			contents: createBoardDTO.contents,
+		});
+		await this.boardsRepository.save(board);
```
create매서드와 save매서드를 이용하여 게시글을 등록해 주었네요. 이방법도 안정적으로 db에 데이터를 저장하는 방법이지만 제생각은  create매서드를 사용하셨다면 save보다는 insert매서드가 나아 보입니다. save매서드는 db에 해당 데이터가 없으면 저장하고 있으면 업데이트 시켜주는 매서드 입니다. 즉 데이터에 접근후 순회하며 검증의 절차를 거치게 됩니다. 이를 방지하기 위해 create와 inserte를 사용한다면 save에서 발생하는 문제를 방지할 수 있습니다. 
읽어 주셔서 감사합니다.

HUMBLE25 commented on src/apis/boards/boards.service.ts:
```diff
@@ -0,0 +1,91 @@
+		const board = await this.getBoardById({ id });
+		board.title = updateBoardDTO.title;
+		board.contents = updateBoardDTO.contents;
+		await this.boardsRepository.save(board);
```
위와같은 맥락으로 저는 save보다는 update매서드가 db를 좀더 효율적으로 다룰수 있을거라 생각합니다.

HUMBLE25 commented on src/apis/boards/boards.service.ts:
```diff
@@ -0,0 +1,91 @@
+			.orderBy({ 'board.createdAt': 'ASC' })
+			.skip(10 * (page - 1))
+			.take(10)
+			.getMany();
```
👍page에따라 게시글을 조회하도록 하셨네요.

HUMBLE25 reviewed (commented):
주말에도 작업하시느라 고생 많으셨습니다.

cabbage556 committed:
feat: save 메서드 수정

createBoard
  - save 메서드를 insert 메서드로 수정

updateBoard
  - save 메서드를 update 메서드로 수정

feat-#9

HUMBLE25 commented on src/apis/boards/boards.service.ts:
```diff
@@ -0,0 +1,91 @@
+			title: createBoardDTO.title,
+			contents: createBoardDTO.contents,
+		});
+		await this.boardsRepository.insert(board);
```
고생하셨습니다 머지하겠습니다.
